src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -80,14 +103,29 @@ public async Task<IActionResult> Validate([FromBody] Resource resource, [FromQue
             }.ToResourceElement());
         }
 
-        [HttpPost]
-        [Route(KnownRoutes.ValidateResourceTypeById)]
-        [AuditEventType(AuditEventSubType.Read)]
-        [Authorize(PolicyNames.ReadPolicy)]
-        [ValidateResourceIdFilter]
-        public async Task<IActionResult> ValidateById([FromBody] Resource resource, [FromQuery(Name = KnownQueryParameterNames.Profile)] string profile, [FromQuery(Name = KnownQueryParameterNames.Mode)] string mode)
+        private static Resource ParseParameters(Parameters resource, ref string profile, ref string mode)
         {
-            return await Validate(resource, profile, mode);
+            var paramMode = resource.Parameter.Find(param => param.Name.Equals("mode", System.StringComparison.OrdinalIgnoreCase));
+            if (paramMode != null && mode != null)
+            {
+                throw new BadRequestException(Resources.MultipleModesProvided);
+            }
+            else if (paramMode != null && mode == null)
+            {
+                mode = paramMode.Value.ToString();
+            }
+
+            var paramProfile = resource.Parameter.Find(param => param.Name.Equals("profile", System.StringComparison.OrdinalIgnoreCase));
+            if (paramProfile != null && profile != null)
+            {
+                throw new BadRequestException(Resources.MultipleProfilesProvided);
+            }
+            else if (paramProfile != null && profile == null)
+            {
+                profile = paramProfile.Value.ToString();
```

Comment:
Shouldn't this throw a `OperationNotImplementedException(Resources.ValidateWithProfileNotSupported)` ?

Reply:
Good catch, I reordered the timing of the check above this so it would catch a profile in the parameters.